### PR TITLE
Added LogglyDestination

### PR DESCRIPTION
**Pull request by xeo-it**

### CODE REVIEW

File: sources/LogglyDestination.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                if let anError = error {
+                    print("Error from Loggly: \(anError)")
+                } else {
+                    if let data = responsedata {
+                        print("Posted to Loggly, status = \(NSString(data: data, encoding:NSUTF8StringEncoding))")
+                    }
+                }
+            })
+        }
+    }
```
Comment (prenagha) [inline sources/LogglyDestination.swift]: Can (should) we hook the flush method in here to have flush push remaining loggly buffer to loggly?
Perhaps have the flush queue message call a flush method on each log destination.

Comment (xeo-it) [inline sources/LogglyDestination.swift]: Not sure I understood it correctly. If you check the buffer is cleared in addLogMsgToBuffer when calling sendLogsInBuffer, and "stringbuffer" is just a local variable that should clear itself once the function returns.   
I've just noticed that if there is an error (loggly not working or anything), the logs are lost so I will implement this in a better way.

Comment (prenagha) [inline sources/LogglyDestination.swift]: Flush tries to make sure queue is emptied before main thread ends. Probably want any buffered loggly items sent to loggly as well as part of flush.

**Comment by skreutzberger** [comment]
Do you have any unit tests, including edge cases like missing / bad internet connection? I am currently writing on something similar but for a different logging service and my solution is somehow much more elaborate (which is not always a good sign, I know). 

I would love SwiftyBeaver to log to Loggly but if we add support for it then it really needs to always work.

**Comment by xeo-it** [comment]
You are right, I'm actually rewriting a bit to improve various edge cases, including bad connection; I've added the rechability class in the project.  
Also, I'm not sure how to handle the case when the lack of internet connection is persistent, so far I save the buffer in memory and send it when the internet connection is available, but if the app shut down before connectivity becomes available, no logs will be ever sent to loggly. 

As I've written in the readme, my solution in not really mine but taken from slimlogger, if your solution is already handling various edge cases, I can hold off for now,  wait for you to finish yours and maybe then apply it to loggly.

**Comment by skreutzberger** [comment]
My solution writes to a buffer file and works with a point system, too. If the point threshold is reached or the app was restarted and there is still content in the buffer file then the content of the file is sent to the server and the file is deleted. The complication goes into what happens if someone is logging while the logs are sent to the server but my solution solves that as well.

I will publish my destination in 1-2 weeks, maybe you can hold off until then or you improve your PR so that it works and then we can compare who has the more elegant solution.

**Comment by xeo-it** [comment]
Ok sounds good, I will give it a go during the weekend and then wait for your solution and compare. 
Thanks.
